catalog/core/catalog-core-dynamic/src/main/java/ddf/catalog/data/dynamic/impl/DynamicMetacard.java

```diff
@@ -0,0 +1,644 @@
+import org.apache.commons.beanutils.DynaClass;
+import org.apache.commons.beanutils.DynaProperty;
+import org.apache.commons.beanutils.LazyDynaBean;
+import org.apache.commons.collections.CollectionUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.AttributeDescriptor;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.MetacardType;
+import ddf.catalog.data.impl.AttributeImpl;
+
+/**
+ * Implements the {@link Metacard}'s required attributes using dynamic beans. This allows for the
+ * creation and modification of beans on the fly.<br/>
+ * <p/>
+ * <p>
+ * <b>Serialization Note</b><br/>
+ * <p/>
+ * <p>
+ * This class is {@link Serializable} and care should be taken with compatibility if changes are
+ * made.
+ * </p>
+ * <p/>
+ * For backward and forward compatibility, {@link ObjectOutputStream#defaultWriteObject()} is
+ * invoked when this object is serialized because it provides "enhanced flexibility" (Joshua Block
+ * <u>Effective Java</u>, Second Edition <i>Item 75</i>). Invoking
+ * {@link ObjectOutputStream#defaultWriteObject()} allows the flexibility to add nontransient
+ * instance fields in later versions if necessary. If earlier versions do not have the newly added
+ * fields, those fields will be ignored and the deserialization will still take place. In addition,
+ * {@link ObjectInputStream#defaultReadObject()} is necessary to facilitate any of the written
+ * fields. </p>
+ * <p/>
+ * <p>
+ * For what constitutes a compatible change in serialization, see <a href=
+ * "http://docs.oracle.com/javase/6/docs/platform/serialization/spec/version.html#6678" >Sun's
+ * Guidelines</a>.
+ * </p>
+ */
+public class DynamicMetacard implements Metacard, MetacardType {
+    private static final Logger LOGGER = LoggerFactory.getLogger(DynamicMetacard.class);
+
+    // the base metacard for dynamic metacards
+    public static final String DYNAMIC = "dynamic";
+
+    // Constants provided by Metacard
+    //public static final String CREATED_DATE = "createdDate"; // -> CREATED
+    //public static final String MODIFIED_DATE = "modifiedDate"; // -> MODIFIED
+    //public static final String EXPIRATION_DATE = "expirationDate";  // -> EXPIRATION
+    //public static final String EFFECTIVE_DATE = "effectiveDate"; // -> EFFECTIVE
+    //public static final String ID = "id";
+    //public static final String THUMBNAIL = "thumbnail";
+    //public static final String TITLE = "title";
+    //public static final String METADATA = "metadata";
+    //public static final String POINT_OF_CONTACT = "pointOfContact";
+    //public static final String DESCRIPTION = "description";
+    //public static final String RESOURCE_URI = "resourceURI";
+    //public static final String RESOURCE_SIZE = "resourceSize";
+    //public static final String SECURITY = "security";
+
+    // Constants used in addition to those from Metacard
+    public static final String LOCATION = "location";
```

Comment:
Yes - removed the commented out Metacard constants. Had them there for reference.